dev/archery/archery/linking.py

```diff
@@ -61,9 +63,84 @@ def list_dependency_names(self):
             names.append(name)
         return names
 
+    def _extract_symbols(self, symbol_info):
+        return [re.search(r'\S+$', line).group() for line in symbol_info if line]
+
+    def _remove_weak_symbols(self, symbol_info):
+        return [line for line in symbol_info if not re.search(r'\s[Ww]\s', line)]
+
+    def _capture_symbols(self, remove_symbol_versions, symbol_info):
+        if remove_symbol_versions:
```

Comment:
With GNU `nm` you can probably use the `--without-symbol-versions` option.

Reply:
It would be with a newer nm version right?

Reply:
Ah, good point, this looks quite recent. Sorry, withdrawing suggestion!